File: register_commands.py
import cmd
import json
import logging
import os
import requests

from dotenv import load_dotenv
load_dotenv()

url = f'https://discord.com/api/v8/applications/{os.getenv("APPLICATION_ID")}/guilds/{os.getenv("DEV_GUILD_ID")}/commands'

logging.basicConfig()
logging.getLogger().setLevel(logging.DEBUG)
logging.debug(f'APPLICATION_ID: {os.getenv("APPLICATION_ID")}')


commands = ""
with open("./lib/commands.json") as cmdfile:
    commands = json.loads(cmdfile.read())

logging.debug(commands)
logging.debug(commands)

response = requests.put(
    url=url,
    headers={
        "Content-Type": "application/json",
        "Authorization": f'Bot {os.getenv("BOT_TOKEN")}',
    },
    json=commands
)

if response.status_code == 200:
    logging.info("Registered all commands")
else:
    logging.error("Error registering commands")
    logging.error(response.text)

File: register_commands.js
const fs = require('node:fs');
const axios = require('axios');

// const commandFiles = fs.readdirSync('./lib/?commands.json');
require('dotenv').config();

(async () => {
  const file = fs.readFileSync('./lib/commands.json');
  const commands = JSON.parse(file);

  console.debug(commands);

  let config = {
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bot ${process.env.BOT_TOKEN}`,
    },
  };
  console.log(commands);
  const response = await axios.put(
    `https://discord.com/api/v8/applications/${process.env.APPLICATION_ID}/guilds/${process.env.GUILD_ID}/commands`,
    commands,
    config
  );

  console.log(response);
  if (response.status == 200) {
    console.info('Registered all commands');
  } else {
    console.error('Error registering commands');
    console.error(response.statusText);
  }
})().catch((e) => {
  console.log(e);
});
